tests/Microsoft.Identity.Test.E2e.netcore/ManagedIdentityImdsTests.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Test.E2E
+{
+    [TestClass]
+    public class ManagedIdentityImdsTests
+    {
+        private const string ArmScope = "https://management.azure.com";
+
+        /// <summary>
+        /// Check if the test is running on Azure Arc.   
+        /// </summary>
+        private static bool IsArc() =>
+            !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("IDENTITY_ENDPOINT"));
+
+        /// <summary>
+        /// Check if the test should run against IMDS.
+        /// </summary>
+        private static bool ShouldRunImds() =>
```

Comment:
There is a simpler approach to this, which is to define an attribute like [RunOnlyOnCI]. Identity.Web has a lot of these. I think MSAL has it too.

Reply:
added now